iTunesUDataPurge/RemoveITunesUCourses.java

```diff
@@ -478,5 +501,186 @@ public static void fileUrl(String fAddress, String localFileName, String destina
+		SOAPMessage soapResponse = soapConnection.call(createLoginRequest(searchServer, username, password), sakaiLoginUrl);
+
+		// print SOAP Response
+		System.out.print("Response SOAP Message:");
+		soapResponse.writeTo(System.out);
+
+		//Parses Soap Response to get session id as string
+		String sessionId = getSessionId(soapResponse);
+		System.out.println("");
+		System.out.println("Session Id:" + sessionId);
+
+		//Test validity of session id by calling Check Session function
+		soapResponse = soapConnection.call(createCheckSessionRequest(searchServer, sessionId), sakaiScriptUrl);
+		System.out.print("Response SOAP Message:");
+		soapResponse.writeTo(System.out);
+
+		String checkedSessionId = getSessionId(soapResponse);
+		System.out.println("");
+		System.out.println("Checked Session Id: " + checkedSessionId);
+
+		//while there are still sites to delete:
+		if(!checkedSessionId.equals("null")){
+			
+			// if session id is not null then remove site
+			soapResponse = soapConnection.call(createRemovePageRequest(searchServer, sessionId, siteId, pageToDelete), sakaiScriptUrl);
+			System.out.print("Response SOAP Message:");
+			soapResponse.writeTo(System.out);
+			System.out.println("");
+			
+		}
+		else{
+			//else get new session id and then process site
+			soapResponse = soapConnection.call(createLoginRequest(searchServer, username, password), sakaiLoginUrl);
+			System.out.print("Response SOAP Message:");
+			soapResponse.writeTo(System.out);
+			
+			checkedSessionId = getSessionId(soapResponse);
+			System.out.println("");
+			System.out.println("Session Id:" + sessionId);
+			
+		}
+		
+		soapConnection.close();
+	}
+
+	public static SOAPMessage createLoginRequest(String serverURI, String username, String password) throws Exception {
+		MessageFactory messageFactory = MessageFactory.newInstance();
+		SOAPMessage soapMessage = messageFactory.createMessage();
+		SOAPPart soapPart = soapMessage.getSOAPPart();
+
+		SOAPEnvelope envelope = soapPart.getEnvelope();
+		envelope.addNamespaceDeclaration("example", serverURI);
+
+		SOAPBody soapBody = envelope.getBody();
+		SOAPElement soapBodyElem = soapBody.addChildElement("login", "example");
+		SOAPElement soapBodyElem1 = soapBodyElem.addChildElement("id", "example");
+		soapBodyElem1.addTextNode(username);
+		SOAPElement soapBodyElem2 = soapBodyElem.addChildElement("pw", "example");
+		soapBodyElem2.addTextNode(password);
+
+		MimeHeaders headers = soapMessage.getMimeHeaders();
+		headers.addHeader("SOAPAction", serverURI  + "login");
+
+		soapMessage.saveChanges();
+
+		/* Print the request message */
+		System.out.print("Request SOAP Message:");
+		soapMessage.writeTo(System.out);
+		System.out.println();
+
+		return soapMessage;
+	}
+
+	public static String getSessionId(SOAPMessage soapResponse ) {
+		String sessionId = "sessionId";
+		try {
+			sessionId = soapResponse.getSOAPBody().getFirstChild().getFirstChild().getTextContent();
```

Comment:
Just the response, I think, because we are parsing the response xml text here. Thanks!